packages/lodestar/src/chain/factory/block/body.ts

```diff
@@ -46,7 +46,7 @@ export async function assembleBody(
   const [proposerSlashings, attesterSlashings, attestations, voluntaryExits, {eth1Data, deposits}] = await Promise.all([
     db.proposerSlashing.values({limit: MAX_PROPOSER_SLASHINGS}),
     db.attesterSlashing.values({limit: MAX_ATTESTER_SLASHINGS}),
-    db.aggregateAndProof.getBlockAttestations(currentState).then((value) => value.slice(0, MAX_ATTESTATIONS)),
+    chain.aggregatedAttestationPool.getAttestationsForBlock(currentState).slice(0, MAX_ATTESTATIONS),
```

Comment:
Instead of slicing here the pool should be aware of MAX_ATTESTATIONS and stop doing work when it gets to MAX_ATTESTATIONS